### PR TITLE
them view cho product va blog

### PR DESCRIPTION
**Pull request by HuyHoang-20130266**

### CODE REVIEW

File: src/main/java/vn/edu/hcmuaf/fit/beans/Blogs.java [hunk 1]
```diff
@@ -145,6 +155,7 @@ public String toString() {
                 ", updateDate='" + updateDate + '\'' +
                 ", catName='" + catName + '\'' +
                 ", catId='" + catId + '\'' +
+                ", ViewCount='" + ViewCount + '\'' +
```
Comment (hoang20130267) [inline src/main/java/vn/edu/hcmuaf/fit/beans/Blogs.java:158]: ok

File: src/main/java/vn/edu/hcmuaf/fit/beans/Product.java [hunk 2]
```diff
@@ -264,6 +274,7 @@ public String toString() {
                 ", quantityCart=" + quantityCart +
                 ", quantityWishlist=" + quantityWishlist +
                 ", cate_id='" + cate_id + '\'' +
+                ", ViewCount='" + ViewCount + '\'' +
```
Comment (hoang20130267) [inline src/main/java/vn/edu/hcmuaf/fit/beans/Product.java:277]: ok

File: src/main/java/vn/edu/hcmuaf/fit/dao/BlogDAO.java [hunk 3]
```diff
@@ -245,6 +245,14 @@ public boolean isCateContainBl(String cateId) {
...
         else return false;
     }
+
+    public void AddViewCountBlog(String id) {
+        JDBIConnector.get().withHandle(handle ->
+                handle.createUpdate("UPDATE blogs SET ViewCount = ViewCount + 1 WHERE BlogId=?")
+                        .bind(0, id)
+                        .execute());
+    }
+
```
Comment (hoang20130267) [inline src/main/java/vn/edu/hcmuaf/fit/dao/BlogDAO.java:255]: ok

File: src/main/java/vn/edu/hcmuaf/fit/API/test.http [hunk 4]
```diff
@@ -0,0 +1,6 @@
+GET http://140.238.54.136/api/district?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vMTQwLjIzOC41NC4xMzYvYXBpL2F1dGgvbG9naW4iLCJpYXQiOjE2ODM3MzEzODYsImV4cCI6MTY4MzczMTk4NiwibmJmIjoxNjgzNzMxMzg2LCJqdGkiOiJPY29nZHhVQ2FxWTRRSkNpIiwic3ViIjoiNTg5ZGY4MDBjZGYyNGZmMWEwM2MwMGYyMTQ4OWYwMDciLCJwcnYiOiIyM2JkNWM4OTQ5ZjYwMGFkYjM5ZTcwMWM0MDA4NzJkYjdhNTk3NmY3In0.BrAoPvPHvJ9oFXbJ2xiBhyvHX4sI_3_VlI4vuMmUA2g
+Content-Type:application/json
+
+{
+  "Province": "202"
+}
```
Comment (hoang20130267) [inline src/main/java/vn/edu/hcmuaf/fit/API/test.http:6]: cái này là gì vậy

File: src/main/java/vn/edu/hcmuaf/fit/dao/ProductDAO.java [hunk 5]
```diff
@@ -468,7 +468,12 @@ public int getQuantityProduct(String idProduct){
                         .bind(0, idProduct).mapTo(Integer.class).first());
         return quantity;
     }
-
+    public void AddViewCountProduct(String id) {
+        JDBIConnector.get().withHandle(handle ->
+                handle.createUpdate("UPDATE product_category SET ViewCount = ViewCount + 1 WHERE CatId=?")
+                        .bind(0, id)
+                        .execute());
+    }
```
Comment (hoang20130267) [inline src/main/java/vn/edu/hcmuaf/fit/dao/ProductDAO.java:476]: ok

File: src/main/java/vn/edu/hcmuaf/fit/services/ProductService.java [hunk 6]
```diff
@@ -61,12 +66,16 @@ public static List<String> getData5() {
...
+    public void AddViewCountProduct(String id) {
+        new ProductDAO().AddViewCountProduct(id);
+    }
 
     public Product getProductDetail(String Id) {
         ProductDAO dao = new ProductDAO();
         return dao.getProductDetail(Id);
     }
 
+
```
Comment (hoang20130267) [inline src/main/java/vn/edu/hcmuaf/fit/services/ProductService.java:78]: ok

File: src/main/webapp/blog-details.jsp [hunk 7]
```diff
@@ -235,6 +236,15 @@
...
                     Blogs blogs = service.getContent(id);
+                    List<String> ViewCount = (List<String>) request.getSession().getAttribute("ViewCount");
+                    if (ViewCount == null) {
+                        ViewCount = new ArrayList<>();
+                        request.getSession().setAttribute("ViewCount", ViewCount);
+                    }
+                    if (!ViewCount.contains(blogs.getBlogId())) {
+                        ViewCount.add(blogs.getBlogId());
+                        BlogService.getInstance().AddViewCountBlog(blogs.getBlogId());
+                    }
```
Comment (hoang20130267) [inline src/main/webapp/blog-details.jsp:247]: ok

File: src/main/webapp/product-details.jsp [hunk 8]
```diff
@@ -315,10 +316,20 @@
...
+                List<ImageProduct> img = service.getListImg(id);
+                List<String> ViewCount = (List<String>) request.getSession().getAttribute("ViewCount1");
+                if (ViewCount == null) {
+                    ViewCount = new ArrayList<>();
+                    request.getSession().setAttribute("ViewCount1", ViewCount);
+                }
+                if (!ViewCount.contains(product.getCate_id())) {
+                    ViewCount.add(product.getCate_id());
+                    ProductService.getInstance().AddViewCountProduct(product.getCate_id());
+                }
```
Comment (hoang20130267) [inline src/main/webapp/product-details.jsp:332]: ok

File: src/main/webapp/product-details.jsp [hunk 9]
```diff
@@ -368,7 +379,7 @@
                 <div class="product__details__text">
                     <h3><%=product.getProductName()%>
                     </h3>
-
+                    <span>Lượt xem: <%=product.getViewCount()%></span>
```
Comment (hoang20130267) [inline src/main/webapp/product-details.jsp:382]: ok

File: target/Petshop-website-final/product-details.jsp [hunk 10]
```diff
@@ -315,10 +316,20 @@
...
+                List<ImageProduct> img = service.getListImg(id);
+                List<String> visiteds = (List<String>) request.getSession().getAttribute("listVisited");
+                if (visiteds == null) {
+                    visiteds = new ArrayList<>();
+                    request.getSession().setAttribute("listVisited", visiteds);
+                }
+                if (!visiteds.contains(product.getCate_id())) {
+                    visiteds.add(product.getCate_id());
+                    ProductService.getInstance().AddViewCountProduct(product.getCate_id());
+                }
```
Comment (hoang20130267) [inline target/Petshop-website-final/product-details.jsp:332]: ok

File: src/main/java/vn/edu/hcmuaf/fit/API/test.http (hunk 4 above)
Comment (HuyHoang-20130266) [reply src/main/java/vn/edu/hcmuaf/fit/API/test.http:6]: test cái api

**Pull request merged**